examples/QuickJournalSync/README.md

```diff
@@ -0,0 +1,386 @@
+    ```csharp
+    var appConfiguration = new AppConfiguration(config.AppId);
+    appConfiguration.SyncTimeoutOptions.ConnectTimeout = TimeSpan.FromSeconds(4);
+    app = Realms.Sync.App.Create(appConfiguration);
+
+    // ....
+
+    try
+    {
+        var configuration = new FlexibleSyncConfiguration(app.CurrentUser)
+        {
+            CancelAsyncOperationsOnNonFatalErrors = true
+        };
+        var realm = await Realm.GetInstanceAsync(configuration);
+    }
+    catch (Exception ex)
+    {
+        Console.WriteLine("GetInstanceAsync timeout");
+    }
+    ```
+    In most cases this approach could be an overkill, as the exception would be raised for any kind of non-fatal errors, even for just a brief connection issue. For this reason it is recommended to use the previous approach, as it allows to retry in the specified time frame.
+
+### Recommended flow
+
+Given the differences between `GetInstance` and `GetInstanceAsync` that we have presented, here is an example of a recommended *logical* flow that uses both methods to open a realm, and that can be used in most applications:
+
+```csharp
+Realm realm;
+
+if (app.CurrentUser == null)
+{
+    // There is no current user, so show the login screen and wait for the user to login.
+    await PresentLoginScreen();
+
+    var realmConfig = GetRealmConfig(app.CurrentUser);
+
+    // Creates a CancellationTokenSource that will be cancelled after 4 seconds.
+    var cts = new CancellationTokenSource(4000);
+
+    try
+    {
+        // We get the realm asynchronously after login, with all the data synchronized.
+        realm = await Realm.GetInstanceAsync(realmConfig, cts.Token);
+    }
+    catch (TaskCanceledException)
+    {
+        // If there are connectivity issues, get the realm synchronously
+        realm = Realm.GetInstance(realmConfig);
+    }
+}
+else
+{
+    var realmConfig = GetRealmConfig(app.CurrentUser);
+
+    // The user is already logged in, so open the realm straight away
+    // with the available data. If the application needs the latest
+    // available data, then use GetInstanceAsync here instead.
+    realm = Realm.GetInstance(realmConfig);
+}
+```
+Please note that this snippet is here only to give an idea about the logical flow of obtaining a realm, but the actual code in your application could look different.
+
+#### How the realm is opened in the sample application
+
+The way that the sample application deals with offline realms loosely follow the snippet for the recommended flow that was shown in the previous section:
+- When the application opens, the `LoginPage` is the first to be loaded.
+- If the user is not logged in already (`CurrentUser == null`), then the user can specify its email and password and login. The login method (`RealmService.LoginAsync`) uses `Realm.GetInstanceAsync`, because we want to be sure to have a completely synchronized realm before showing the main page to the user:
```

Comment:
```suggestion
- If the user is not logged in already (`CurrentUser == null`), then the user can specify their email and password and login. The login method (`RealmService.LoginAsync`) uses `Realm.GetInstanceAsync`, because we want to be sure to have a completely synchronized realm before showing the main page to the user:
```